core/standalone/src/main/scala/org/apache/openwhisk/standalone/UserEventLauncher.scala

```diff
@@ -0,0 +1,148 @@
+    } yield {
+      logging.info(this, "Enabled the user-event config")
+      System.setProperty("whisk.user-events.enabled", "true")
+      Seq(userEvent, promSvc, grafanaSvc)
+    }
+  }
+
+  def runUserEvents(): Future[ServiceContainer] = {
+    val env = Map("KAFKA_HOSTS" -> s"$hostIp:$kafkaDockerPort")
+
+    logging.info(this, s"Starting User Events: $userEventPort")
+    val name = containerName("user-events")
+    val params = Map("-p" -> Set(s"$userEventPort:9095"))
+    val args = createRunCmd(name, env, params)
+
+    val f = docker.runDetached(userEventConfig.image, args, true)
+    f.map(_ => ServiceContainer(userEventPort, s"http://localhost:$userEventPort", name))
+  }
+
+  def runPrometheus(): Future[(StandaloneDockerContainer, ServiceContainer)] = {
+    logging.info(this, s"Starting Prometheus at $prometheusPort")
+    val baseParams = Map("-p" -> Set(s"$prometheusPort:9090"))
+    val promConfigDir = newDir(workDir, "prometheus")
+    val promDataDir = newDir(dataDir, "prometheus")
+
+    val configFile = new File(promConfigDir, "prometheus.yml")
+    FileUtils.write(configFile, prometheusConfig, UTF_8)
+
+    val volParams = Map(
+      "-v" -> Set(s"${promDataDir.getAbsolutePath}:/prometheus", s"${promConfigDir.getAbsolutePath}:/etc/prometheus/"))
+    val name = containerName("prometheus")
+    val args = createRunCmd(name, Map.empty, baseParams ++ volParams)
+    val f = docker.runDetached(userEventConfig.prometheusImage, args, shouldPull = true)
+    val sc = ServiceContainer(prometheusPort, s"http://localhost:$prometheusPort", name)
+    f.map(c => (c, sc))
+  }
+
+  def runGrafana(promContainer: StandaloneDockerContainer): Future[ServiceContainer] = {
+    logging.info(this, s"Starting Grafana at $grafanaPort")
+    val baseParams = Map("-p" -> Set(s"$grafanaPort:3000"))
+    val grafanaConfigDir = newDir(workDir, "grafana")
+    val grafanaDataDir = newDir(dataDir, "grafana")
+
+    val promUrl = s"http://$hostIp:$prometheusPort"
+    unzipGrafanaConfig(grafanaConfigDir, promUrl)
+
+    val env = Map(
+      "GF_PATHS_PROVISIONING" -> "/etc/grafana/provisioning",
+      "GF_USERS_ALLOW_SIGN_UP" -> "false",
+      "GF_AUTH_ANONYMOUS_ENABLED" -> "true",
+      "GF_AUTH_ANONYMOUS_ORG_NAME" -> "Main Org.",
+      "GF_AUTH_ANONYMOUS_ORG_ROLE" -> "Admin")
+
+    val volParams = Map(
+      "-v" -> Set(
+        s"${grafanaDataDir.getAbsolutePath}:/var/lib/grafanas",
+        s"${grafanaConfigDir.getAbsolutePath}/provisioning/:/etc/grafana/provisioning/",
+        s"${grafanaConfigDir.getAbsolutePath}/dashboards/:/var/lib/grafana/dashboards/"))
+    val name = containerName("grafana")
+    val args = createRunCmd(name, env, baseParams ++ volParams)
+    val f = docker.runDetached(userEventConfig.grafanaImage, args, shouldPull = true)
+    val sc = ServiceContainer(grafanaPort, s"http://localhost:$grafanaPort", name)
+    f.map(_ => sc)
+  }
+
+  private def prometheusConfig =
+    s"""global:
+  |  scrape_interval: 10s
```

Comment:
The scrape interval definition has to be matched with the `rate` function interval (at least in the current dashboards). I wonder if we should not import this configuration from: `/compose/prometheus/prometheus.yml`

Reply:
Fixed that by packaging the existing `prometheus.yml` and then modifying it